crates/bevy_transform/src/hierarchy/child_builder.rs

```diff
@@ -282,6 +284,43 @@ impl<'w> BuildWorldChildren for EntityMut<'w> {
         self.update_location();
         self
     }
+
+    fn push_children(&mut self, children: &[Entity]) -> &mut Self {
+        let parent = self.id();
+        // SAFETY: update_location is called after the entity is modified
+        let world = unsafe { self.world_mut() };
+        for child in children.iter() {
+            world
+                .entity_mut(*child)
+                .insert_bundle((Parent(parent), PreviousParent(parent))); // FIXME: dom't erase the previous parent (see #1545)
+        }
+        if let Some(mut children_component) = world.get_mut::<Children>(parent) {
+            children_component.0.extend(children.iter().cloned());
+        } else {
+            world.entity_mut(parent).insert(Children::with(children));
```

Comment:
We can optimize this by using self (which is `world.entity_mut(parent)`). Additionally, we haven't updated the entity location, so theres no need to call self.update_location() here.

I'll just push the changes so we can get this merged asap.

Reply:
I guess I should say, "we haven't modified the entity location using the unsafe world reference". Using self.insert will handle the location stuff for us.